cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1861,11 +1861,15 @@ func (p *BicepProvider) ensureParameters(
 		param := template.Parameters[key]
 
 		// If a value is explicitly configured via a parameters file, use it.
+		// unless the parameter value inference is nil/empty
 		if v, has := parameters[key]; has {
-			configuredParameters[key] = azure.ArmParameterValue{
-				Value: armParameterFileValue(p.mapBicepTypeToInterfaceType(param.Type), v.Value),
+			paramValue := armParameterFileValue(p.mapBicepTypeToInterfaceType(param.Type), v.Value)
+			if paramValue != nil {
```

Comment:
What do you think about letting `armParameterFileValue` to return empty string as before and then, adding here a condition that checks if the `key` is set on the `environment`.  If yes, we use the empty string (backwards compatibility), and if no, we don't add the the parameter to `configuredParameters` (like you have here)

Reply:
Also, the `envsubst.Eval()` package that is in use also supports more complex expressions where you could even referenced multiple environment variables in the same expression.

I still feel that this is a pretty safe change - if the expression evaluates to an empty string and the parameter does not have a default then we prompt the user to supply a value..